File: app/src/main/java/com/artamonov/millionplanets/fight/presenter/FightActivityPresenter.java
package com.artamonov.millionplanets.fight.presenter;

import com.artamonov.millionplanets.fight.FightActivityView;
import com.artamonov.millionplanets.model.User;
import com.google.firebase.firestore.DocumentSnapshot;

public interface FightActivityPresenter<V extends FightActivityView> {

    User getUserList();

    void calculateDamage();

    void calculateDamageFromEnemy();

    void calculateDamageToEnemy();

    void calculateLoot();

    Boolean fightFinished();

    void setLootTransferFinished(Boolean state);

    void setUserList(DocumentSnapshot doc);

    void setEnemyList(DocumentSnapshot doc);
}

File: app/src/main/java/com/artamonov/millionplanets/fight/presenter/FightActivityPresenter.kt
package com.artamonov.millionplanets.fight.presenter

import com.artamonov.millionplanets.fight.FightActivityView
import com.artamonov.millionplanets.model.User
import com.google.firebase.firestore.DocumentSnapshot

interface FightActivityPresenter<V : FightActivityView?> {
    fun getUserList(): User
    fun calculateDamage()
    fun calculateDamageFromEnemy()
    fun calculateDamageToEnemy()
    fun calculateLoot()
    fun fightFinished(): Boolean
    fun setLootTransferFinished(state: Boolean)
    fun setUserList(doc: DocumentSnapshot)
    fun setEnemyList(doc: DocumentSnapshot)
}